golem-component-service-base/src/repo/component.rs

```diff
@@ -419,12 +632,13 @@ impl ComponentRepo for DbComponentRepo<sqlx::Postgres> {
             let namespace: String = result.get("namespace");
             let name: String = result.get("name");
             if namespace != component.namespace || name != component.name {
+                transaction.rollback().await?;
```

Comment:
is there a higher level transaction construct in sqlx? manually doing the rollbacks on errors seems fragile

Reply:
I would change this select to be part of the insert (or rather, an upsert, and check if there were any affected rows, otherwise i think there are some race conditions here.

Reply:
It should be rolled back on drop, my tests indicated that it is was not working properly, and adding an explicit rollback fixed the test. I don't like it either.